From 5225bf0e8afa7ee51a1f90152f68199864bb4b52 Mon Sep 17 00:00:00 2001
From: Jose Luis Blanco Claraco <joseluisblancoc@gmail.com>
Date: Fri, 18 May 2018 14:31:20 +0200
Subject: [PATCH] dont build most mrpt apps (save build time)

---
 debian/rules.em | 47 ++++++++++++++++++++++++++++++++++++++++++++++-
 1 file changed, 46 insertions(+), 1 deletion(-)

diff --git a/debian/rules.em b/debian/rules.em
index aa010fa..e9d30d6 100755
--- a/debian/rules.em
+++ b/debian/rules.em
@@ -26,7 +26,52 @@ override_dh_auto_configure:
 	# in the install tree that was dropped by catkin, and source it.  It will
 	# set things like CMAKE_PREFIX_PATH, PKG_CONFIG_PATH, and PYTHONPATH.
 	if [ -f "@(InstallationPrefix)/setup.sh" ]; then . "@(InstallationPrefix)/setup.sh"; fi && \
-	dh_auto_configure -- -DCMAKE_INSTALL_PREFIX="@(InstallationPrefix)"
+	dh_auto_configure -- -DCMAKE_INSTALL_PREFIX="@(InstallationPrefix)" \
+	-DBUILD_APP_2d-slam-demo=OFF \
+	-DBUILD_APP_DifOdometry-Camera=OFF \
+	-DBUILD_APP_DifOdometry-Datasets=OFF \
+	-DBUILD_APP_GridmapNavSimul=OFF \
+	-DBUILD_APP_RawLogViewer=ON \
+	-DBUILD_APP_ReactiveNav3D-Demo=OFF \
+	-DBUILD_APP_ReactiveNavigationDemo=OFF \
+	-DBUILD_APP_SceneViewer3D=ON \
+	-DBUILD_APP_camera-calib=ON \
+	-DBUILD_APP_carmen2rawlog=OFF \
+	-DBUILD_APP_carmen2simplemap=OFF \
+	-DBUILD_APP_features-matching=OFF \
+	-DBUILD_APP_gps2rawlog=ON \
+	-DBUILD_APP_graph-slam=ON \
+	-DBUILD_APP_graphslam-engine=OFF \
+	-DBUILD_APP_grid-matching=OFF \
+	-DBUILD_APP_hmt-slam=OFF \
+	-DBUILD_APP_hmt-slam-gui=OFF \
+	-DBUILD_APP_hmtMapViewer=OFF \
+	-DBUILD_APP_holonomic-navigator-demo=OFF \
+	-DBUILD_APP_icp-slam=ON \
+	-DBUILD_APP_icp-slam-live=ON \
+	-DBUILD_APP_image2gridmap=ON \
+	-DBUILD_APP_kf-slam=OFF \
+	-DBUILD_APP_kinect-3d-slam=OFF \
+	-DBUILD_APP_kinect-3d-view=OFF \
+	-DBUILD_APP_kinect-stereo-calib=OFF \
+	-DBUILD_APP_map-partition=OFF \
+	-DBUILD_APP_mrpt-perfdata2html=OFF \
+	-DBUILD_APP_mrpt-performance=OFF \
+	-DBUILD_APP_navlog-viewer=ON \
+	-DBUILD_APP_observations2map=ON \
+	-DBUILD_APP_pf-localization=OFF \
+	-DBUILD_APP_ptg-configurator=ON \
+	-DBUILD_APP_rawlog-edit=ON \
+	-DBUILD_APP_rawlog-grabber=ON \
+	-DBUILD_APP_rbpf-slam=ON \
+	-DBUILD_APP_ro-localization=OFF \
+	-DBUILD_APP_robotic-arm-kinematics=OFF \
+	-DBUILD_APP_simul-beacons=OFF \
+	-DBUILD_APP_simul-gridmap=OFF \
+	-DBUILD_APP_simul-landmarks=OFF \
+	-DBUILD_APP_track-video-features=OFF \
+	-DBUILD_APP_velodyne-view=ON
+
 
 override_dh_auto_build:
 	# In case we're installing to a non-standard location, look for a setup.sh
-- 
2.17.1

